From c563c0b8d2a3f08921e2c2e87528801bf3abeb76 Mon Sep 17 00:00:00 2001
From: Zhong Jinghua <zhongjinghua@huawei.com>
Date: Mon, 10 Oct 2022 17:18:46 +0800
Subject: [PATCH 0598/1076] [Huawei] Revert "[Huawei] quota: Check next/prev
 free block number after reading from quota file"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187046

--------------------------------

This reverts commit 5c9e489cb1d0bf07de4538250ffdc23d5cc0c62b.

This patch is temporary, revert it and backport the mainline patch.

Signed-off-by: Zhong Jinghua <zhongjinghua@huawei.com>
---
 fs/quota/quota_tree.c | 38 --------------------------------------
 1 file changed, 38 deletions(-)

diff --git a/fs/quota/quota_tree.c b/fs/quota/quota_tree.c
index 15b8e1a39015..1a188fbdf34e 100644
--- a/fs/quota/quota_tree.c
+++ b/fs/quota/quota_tree.c
@@ -80,35 +80,6 @@ static ssize_t write_blk(struct qtree_mem_dqinfo *info, uint blk, char *buf)
 	return ret;
 }
 
-static inline int do_check_range(struct super_block *sb, uint val, uint max_val)
-{
-	if (val >= max_val) {
-		quota_error(sb, "Getting block too big (%u >= %u)",
-			    val, max_val);
-		return -EUCLEAN;
-	}
-
-	return 0;
-}
-
-static int check_free_block(struct qtree_mem_dqinfo *info,
-			    struct qt_disk_dqdbheader *dh)
-{
-	int err = 0;
-	uint nextblk, prevblk;
-
-	nextblk = le32_to_cpu(dh->dqdh_next_free);
-	err = do_check_range(info->dqi_sb, nextblk, info->dqi_blocks);
-	if (err)
-		return err;
-	prevblk = le32_to_cpu(dh->dqdh_prev_free);
-	err = do_check_range(info->dqi_sb, prevblk, info->dqi_blocks);
-	if (err)
-		return err;
-
-	return err;
-}
-
 /* Remove empty block from list and return it */
 static int get_free_dqblk(struct qtree_mem_dqinfo *info)
 {
@@ -123,9 +94,6 @@ static int get_free_dqblk(struct qtree_mem_dqinfo *info)
 		ret = read_blk(info, blk, buf);
 		if (ret < 0)
 			goto out_buf;
-		ret = check_free_block(info, dh);
-		if (ret)
-			goto out_buf;
 		info->dqi_free_blk = le32_to_cpu(dh->dqdh_next_free);
 	}
 	else {
@@ -273,9 +241,6 @@ static uint find_free_dqentry(struct qtree_mem_dqinfo *info,
 		*err = read_blk(info, blk, buf);
 		if (*err < 0)
 			goto out_buf;
-		*err = check_free_block(info, dh);
-		if (*err)
-			goto out_buf;
 	} else {
 		blk = get_free_dqblk(info);
 		if ((int)blk < 0) {
@@ -468,9 +433,6 @@ static int free_dqentry(struct qtree_mem_dqinfo *info, struct dquot *dquot,
 		goto out_buf;
 	}
 	dh = (struct qt_disk_dqdbheader *)buf;
-	ret = check_free_block(info, dh);
-	if (ret)
-		goto out_buf;
 	le16_add_cpu(&dh->dqdh_entries, -1);
 	if (!le16_to_cpu(dh->dqdh_entries)) {	/* Block got free? */
 		ret = remove_free_dqentry(info, buf, blk);
-- 
2.17.1

